OpenRA.Mods.Common/Widgets/ObserverProductionIconsWidget.cs

```diff
@@ -73,13 +99,20 @@ public override void Draw()
 				.Where(a => a.Actor.Owner == player)
 				.Select((a, i) => new { a.Trait, i });
 
+			if (renderBounds != RenderBounds)
+			{
+				renderBounds = RenderBounds;
+				InitIcons(renderBounds);
+			}
+			else
+				for (var i = 0; i < icons.Length; i++)
+					icons[i] = null;
```

Comment:
You can cut down memory churn a bit more by keeping the `ProductionIcon` but nulling out the `.Actor`.  In `Tick` you can then check if that is null or not to know whether to set `TooltipIcon`.